src/__tests__/getMarkupFromTree-test.tsx

```diff
@@ -0,0 +1,506 @@
+import ApolloClient from 'apollo-client';
+import { ApolloLink, Observable } from 'apollo-link';
+import { MockedResponse } from 'apollo-link-mock';
+import gql from 'graphql-tag';
+import * as React from 'react';
+import { renderToString } from 'react-dom/server';
+
+import { ApolloProvider } from '../ApolloContext';
+import { TestErrorBoundary } from '../__testutils__/TestErrorBoundary';
+import createClient from '../__testutils__/createClient';
+import { getMarkupFromTree } from '../getMarkupFromTree';
+import { QueryHookOptions, useQuery } from '../useQuery';
+
+const AUTH_QUERY = gql`
+  {
+    isAuthorized
+  }
+`;
+
+interface AuthQueryResult {
+  isAuthorized: boolean;
+}
+
+const USER_QUERY = gql`
+  {
+    currentUser {
+      firstName
+    }
+  }
+`;
+
+interface UserQueryResult {
+  currentUser: { firstName: string };
+}
+
+const MOCKS: MockedResponse[] = [
+  {
+    request: { query: AUTH_QUERY },
+    result: { data: { isAuthorized: true } },
+  },
+  {
+    request: { query: USER_QUERY },
+    result: { data: { currentUser: { firstName: 'James' } } },
+  },
+];
+
+const linkReturningError = new ApolloLink(
+  () =>
+    new Observable(observer => {
+      observer.error(new Error('Simulating network error'));
+    })
+);
+
+function createMockClient(link?: ApolloLink) {
+  return createClient({ link, mocks: MOCKS, addTypename: false });
+}
+
+function useAuthDetails(options?: QueryHookOptions<{}>) {
+  const { data, loading } = useQuery<AuthQueryResult>(AUTH_QUERY, options);
+
+  return Boolean(!loading && data && data.isAuthorized);
+}
+
+function UserDetails(props: QueryHookOptions<{}>) {
+  const { data, loading } = useQuery<UserQueryResult>(USER_QUERY, props);
+
+  return (
+    <>
+      {loading
+        ? 'Loading'
+        : !data
+        ? 'No Data (skipped)'
+        : !data.currentUser
+        ? 'No Current User (failed)'
+        : data.currentUser.firstName}
+    </>
+  );
+}
+
+interface UserWrapperProps extends QueryHookOptions<{}> {
+  readonly client: ApolloClient<object>;
+}
+
+function UserDetailsWrapper({ client, ...props }: UserWrapperProps) {
+  return (
+    <TestErrorBoundary>
```

Comment:
What's the purpose of using `<TestErrorBoundary>` here? Tests are passing also without it.

Reply:
I wanted to test this block https://github.com/trojanowski/react-apollo-hooks/blob/2425cfa86cf0b92c327927fdfb36ac7eb1a32559/src/getMarkupFromTree.tsx#L25-L31

But I guess I found the better way and forgot to remove 🤔